include/sys/dnode.h

```diff
@@ -145,6 +145,57 @@ enum dnode_dirtycontext {
 
 #define	DNODE_CRYPT_PORTABLE_FLAGS_MASK		(DNODE_FLAG_SPILL_BLKPTR)
 
+/*
+ * VARIABLE-LENGTH (LARGE) DNODES
```

Comment:
Would it be worth mention the difference between, free, allocated, and interior dnodes?

Reply:
Let's leave this for another time.